### ISSUE TITLE
"pad_start" changes with "tempo" setting

### ISSUE THREAD

**Issue by codemann8** [qwertymodo/msupcmplusplus#5]
It seems that when you have a "pad_start" within the same track as "tempo", the resulting padding doesn't honor the value you set for it.  This should be a fixed value that doesn't change when the tempo is affected. To reproduce:

```json
"sub_channels": [
    {
        "file": "music.mp3",
        "pad_start": 22050,
        "tempo": 0.88,
        "fade_in": 11025,
        "trim_start": 211388,
        "trim_end": 423326
    }
]
```

The short-term workaround is to move the padding to an outside layer and sub_track the remaining track, although undesirable for simplicity sake:

```json
"sub_channels": [
    {
        "pad_start": 22050,
        "file": "music.mp3",
        "sub_tracks": [
            {
                "tempo": 0.88,
                "fade_in": 11025,
                "trim_start": 211388,
                "trim_end": 423326
            }
        ]
    }
]
```

**Comment by qwertymodo** [comment]
Ok, so this one comes down to the order of operations, and I need to give some thought to why I currently have it set the way it is.  The current order of operations is:

- compression
- loop cross-fade mixing
- trim
- normalization
- fading
- padding
- tempo
- loop point recalculation
- dithering

It's been awhile since I last messed with the order of operation, so I don't remember all of the reasons behind it, but here's what I do recall.

Cross-fading needs to happen before trimming, because it uses parts of the audio that will get thrown away by the trim.

Trimming needs to happen before normalization, because you don't want parts of the audio that are discarded by the trim to affect the normalization averaging.  Likewise, padding needs to occur after normalization, so the added silence also doesn't affect it.

Fading happens after trimming, so the fade is applied in the correct place.

I need to think about it a bit more, but off the top of my head, I can't think of any reason why tempo can't be moved up one, so it's immediately before padding.  That would fix the issue here.  As a workaround, you can always force an order-of-operation change by further nesting the sub_tracks and sub_channels, as the recursive execution is depth-first, and each nested level is fully rendered before passing it as an input file into the parent.

**Comment by qwertymodo** [comment]
Should be fixed in 89368c4cb98a1b8d249ace0096ec8f6fd24cd45e
